WISE-Community/WISE-Client
# refactor(RegisterUserFormComponent): Extract duplicate createAccountError()

## Changes
- Extract ```Register[Student/Teacher]FormComponent.createAccountError()``` to parent ```RegisterUserFormComponent.createAccountError()```
- Clean up surrounding code
   - use ```this.user``` instead of ```this.studentUser``` and ```this.teacherUser``` in the components
   - move ```processing``` and ```passwordsFormGroup``` variables to ```RegisterUserFormComponent```

## Test
- Send an invalid password to the server
   - Set a breakpoint in StudentService.registerStudentAccount() and change the studentUser's password to "1" before it is POST'ed to the server. => verify that RegisterUserFormComponent.createAccountError() is called, and it shows the error on the form
   - Do the same as above with TeacherService.registerTeacherAccount() when registering a new teacher user
- Creating student/teacher accounts without any problems work as before
